config/cmake/addGitHooks.cmake

```
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 2.8.7)
+
+find_package(Git)
+if(NOT GIT_FOUND)
+    message(FATAL_ERROR "git not found!")
+endif()
+
+include(clang-cxx-dev-tools)
+
+function(set_git_hooks_enabled)
+    execute_process(
+            COMMAND ${GIT_EXECUTABLE} config --local core.hooksPath .githooks
+    )
+endfunction()
+
+function(set_git_hooks_disabled)
+    execute_process(
+            COMMAND ${GIT_EXECUTABLE} config --local --unset core.hooksPath
+    )
+endfunction()
+
+option(ENABLE_GIT_HOOKS "Activate git hooks to run clang-format on committed files." ON)
```

I think this should use cmake_dependent_option to make the option default to on when clang-format is available, or off if it isn't.

```suggestion
cmake_dependent_option(ENABLE_GIT_HOOKS "Activate git hooks to run clang-format on committed files." ON "CLANG_FORMAT" OFF)
```